### PR TITLE
[REVIEW] Fix for reading parquet consecutive rowgroups

### LINKED ISSUE #4584: [BUG] read_parquet(), reading consecutive rowgroups through row_group_count argument gives wrong output
**Describe the bug**
On the script below a parquet file is generated with two rowgroups. However, when we set the row_group_count argument in order to read two consecutive rowgroups, cudf gives inconsistent results.

**Steps/Code to reproduce bug**
```
import cudf
import numpy as np

n = 1999999
out_df = cudf.DataFrame({'c1': np.random.choice(range(1000000, 2000000), n), 
                   'c2': np.random.choice(range(20), n), 
                   'c3': np.round(np.random.uniform(1, 5000, n), 2)})

out_df.to_parquet('temp') #parquet file will have two rowgroups

#reading all two rowgroups
df = cudf.read_parquet('temp')
print(df.count())
print(df.tail(10))

#reading consecutive rowgroups starting from rowgroup 0 gives wrong rows
df = cudf.read_parquet('temp', row_group=0, row_group_count=2)
print(df.count())
print(df.tail(10))
```

**Expected behavior**
Both outputs must be the same.
Actual output:
```
c1    1999999
c2    1999999
c3    1999999
dtype: int64
              c1  c2       c3
1999989  1334142  13  3843.63
1999990  1438323   9  2621.67
1999991  1392994   8   544.91
1999992  1895706  16   723.95
1999993  1578448  18   135.85
1999994  1401037  14  4815.32
1999995  1894110  14  4000.29
1999996  1414036  16  3722.96
1999997  1707059   1   202.65
1999998  1965677   6    20.74
c1    1999999
c2    1999999
c3    1999999
dtype: int64
         c1  c2   c3
1999989   0   0  0.0
1999990   0   0  0.0
1999991   0   0  0.0
1999992   0   0  0.0
1999993   0   0  0.0
1999994   0   0  0.0
1999995   0   0  0.0
1999996   0   0  0.0
1999997   0   0  0.0
1999998   0   0  0.0
```

**Environment overview**
 - Environment location: Bare-metal
 - Method of cuDF install: from source branch-0.13

### PR DESCRIPTION
**Pull request by rommelDB**
Fixes #4584

### CODE REVIEW

File: cpp/src/io/parquet/reader_impl.cu [hunk 1]
```diff
@@ -260,7 +260,7 @@ struct metadata : public FileMetaData {
       }
       row_count = 0;
       do {
-        selection.emplace_back(row_group, row_start);
+        selection.emplace_back(row_group, row_count);
```
Comment (OlivierNV) [inline cpp/src/io/parquet/reader_impl.cu]: Should be row_start + row_count

Comment (rommelDB) [reply cpp/src/io/parquet/reader_impl.cu]: Thanks @OlivierNV, that change fixed the Python tests.

**Comment by vuule** [comment]
Is there a test that fails without this fix?

**Comment by OlivierNV** [comment]
> Is there a test that fails without this fix?

No (largely because it was added before the parquet writer was ported to the non-legacy path). We could now add a test for this in test_parquet.py, though it presumably will get this coverage through dask parquet tests.

**Pull request merged**